src/Serenity.Scripts/CoreLib/UI/Helpers/UploadHelper.ts

```diff
@@ -23,8 +23,12 @@ export namespace UploadHelper {
             pasteZone: options.zone,
             done: function (e: JQueryEventObject, data: any) {
                 var response = data.result;
-                if (options.fileDone != null) {
-                    options.fileDone(response, data.files[0].name, data);
+                if (response.Error != null) {

```

Comment:
using `if (!response.Error)` will be eliminate some of misleading exceptions when Error is undefined or empty string etc.